### PR TITLE
Confinement

### PR DESCRIPTION
**Pull request by KellyMWhitehead**
Matt, I set up confinement in the GNAT package. This can be moved or modified as needed.

Let me know if I anything sticks out as odd or bad form.

### CODE REVIEW

File: packages/gnat/.vscode/launch.json [hunk 1]
```diff
@@ -38,6 +38,25 @@
...
+    {
+      "name": "Confinement",
+      "type": "python",
+      "request": "launch",
+      "module": "confinement.confinement",
+      "console": "integratedTerminal",
+      // Add NATIONAL_PROJECT, INPUTS_FOLDER, DOWNLOAD_FOLDER to your .env file
+      "args": [
+        "${input:HUC}",
+        "D:\\NAR\\Projects\\Confinement\\flowlines.shp",
```
Comment (MattReimer) [inline packages/gnat/.vscode/launch.json:51]: Totally minor thing but I wouldn't leave a hardcoded path in here. Use the .env file (take a look at rs_context for examples)

File: packages/gnat/confinement/confinement.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    # TODO : clean/test outputs?
+
+    # Save Outputs to Geopackage
+
+    out_driver = ogr.GetDriverByName("GPKG")
+    data_out = out_driver.Open(output_gpkg, 1)
+    lyr_out_confining_margins = data_out.CreateLayer('ConfiningMargins', srs, geom_type=ogr.wkbLineString)
+    feature_def = lyr_out_confining_margins.GetLayerDefn()
+    feature = ogr.Feature(feature_def)
+    feature.SetGeometry(ogr.CreateGeometryFromJson(json.dumps(mapping(geom_confining_margins))))
```
Comment (MattReimer) [inline packages/gnat/confinement/confinement.py:118]: Take a look at this:

https://github.com/Riverscapes/riverscapes-tools/blob/master/lib/commons/rscommons/shapefile.py#L876

Going to and from wkb is less memory and less steps.

**Review by MattReimer — CHANGES_REQUESTED** [review]

**Review by MattReimer — APPROVED** [review]
Looks good to me. When you're ready to scale this up we can try the tool on a large number of hucs to see if there are issues we can't see at present.

**Review by MattReimer — APPROVED** [review]

**Comment by philipbaileynar** [comment]
@MattReimer I just pushed the confinement report. It's super super crude right now. But it's a placeholder to take input from folks.

It required a little refactoring of the common reporting code so I ran BRAT Run once to ensure that I hadn't broken anything.

**Review by MattReimer — APPROVED** [review]

**Review by MattReimer — APPROVED** [review]

**Pull request merged**